### DISCUSSION TITLE
Cannot declare parameters to receive multiples values with pydantic BaseModel and Depends

### DISCUSSION THREAD

**Discussion by berlotto** [fastapi/fastapi#8827] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional, List

from fastapi import FastAPI, Depends, Query
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None
    relates_to: List[int] = Query(...)


@app.get("/items/")
async def sample_get_items(
    item: Item = Depends()
):
    return "ok"
```

### Description

This code is based on the samples in the fastapi site. 

The problem is in the parameter `relates_to` . This parameter not appears in the /docs as a parameter because this is of the tytpe `List[int]` 

I need, by this way, declaring parameters for a endpoint thought a pydantic.BaseModel, that have possibility to receive multiple values.

But the way it's written in the example, it doesn't work, either by sending 
`/item/?relates_to=10&relates_to=11&relates_to=12` or `/item/?relates_to=[10,11,12]` or other way...

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.9.5

**Comment by trevorwang** [comment]
The sample code defined a Class with `__init__` method, but you didn't

```python
commons: CommonQueryParams = Depends()

```

```python

class CommonQueryParams:
    def __init__(self, q: Optional[str] = None, skip: int = 0, limit: int = 100):
        self.q = q
        self.skip = skip
        self.limit = limit
```

**Comment by berlotto** [comment]
@trevorwang What page of the documentation shows a sample like this code you wrote ? I didn't see this

But, how this __init__ will help me ?

**Comment by trevorwang** [comment]
Please refer https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/?h=commonqueryparams

**Comment by berlotto** [comment]
Ok, I got it, but, in this way, declaring the `__init__ ` method, I will to change some classes of input data because one type of parameter do not works like the others

An Input classe with ONE parameter that accept multiple values need to be:

```python
class Item:
    def __init__(self, 
        name: str,
        price: float,
        description: Optional[str] = None,
        tax: Optional[float] = None,
        relates_to: List[int] = Query(...)
    ):
        self.name = name
        self.price = price
        self.description = description
        self.tax = tax
        self.relates_to = relates_to
```

rather than

```python
class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None
    relates_to: List[int] = Query(...)
```

That makes sense ?

**Comment by trevorwang** [comment]
I just showed you the supported way to accept multiple parameters.
But I agree with you. A class extending BaseModel is a better choice.

**Comment by berlotto** [comment]
How this #question can be transformed in a #issue ?

**Comment by trevorwang** [comment]
have no idea about that.  o(╥﹏╥)o

**Comment by Kludex** [comment]
You are not supposed to have a `Query` inside the Pydantic model.

> How this #question can be transformed in a #issue ?

Labels are defined by @tiangolo .

**Comment by berlotto** [comment]
> You are not supposed to have a `Query` inside the Pydantic model.
> 

I undertand, but is the only way that this works in a Pydantic model (see image) to be simple

![image](https://user-images.githubusercontent.com/5214/136128069-c3a4d548-84b1-429c-9ea8-9a1e3ec86cb8.png)

**Comment by YuriiMotov** [comment, marked as answer]
Since FastAPI 0.115.0 the following works:

```python
from typing import List, Optional

from fastapi import FastAPI, Query
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None
    relates_to: List[int]


@app.get("/items/")
async def sample_get_items(item: Item = Query()):
    return "ok"

```
(just changed `Depends` to `Query`)

Read more: https://fastapi.tiangolo.com/tutorial/query-param-models/
